src/unix/apple/disk.rs

```diff
@@ -92,7 +92,11 @@ impl crate::DisksInner {
 
     pub(crate) fn refresh_list(&mut self) {
         unsafe {
-            get_list(&mut self.disks);
+            // SAFETY: We don't keep any Objective-C objects around because we 
+            // don't make any direct Objective-C calls in this code.
+            AutoreleasePool::new_with(|| {
```

Comment:
At this point, please just make it a standalone function taking a callback (I really like this idea!). No need to have a struct in this context.

Reply:
Sure. Please have a look at the latest revision.